pkg/types/types.go

```diff
@@ -11,6 +11,13 @@ type VfState struct {
 	EffectiveMAC string `json:"effective_mac"`
 }
 
+// Trunk represents configuration options for VLAN trunk
+type Trunk struct {
+	MinID *int `json:"minID,omitempty"`
+	MaxID *int `json:"maxID,omitempty"`
+	ID    *int `json:"id,omitempty"`
+}
+
```

Comment:
nit: you could move the validation logic for a single trunk entry as a member method of Trunk